### DISCUSSION TITLE
Cluster statistic

### DISCUSSION THREAD

**Discussion by rsarkisov82** [ant-media/Ant-Media-Server#6321] [Q&A] [answered]
We have an ANTmedia cluster with multiple Origin and Edge nodes and a Load Balancer. Is there a way, perhaps using API or logging options, to obtain the list of published source streams associated with their Origin nodes and/or the list of streams with Edge nodes? In other words, can we create a comprehensive overview of the inbound and outbound streams within the cluster?

**Comment by yashtandon113** [comment]
@rsarkisov82 
Thanks for asking the question.

You can use the [get broadcast list](https://antmedia.io/rest/#/default/getBroadcastList) API to get the stream data. You will get the response as below, in which you can see the node IP address (Origin or Edge) on which it was created. Check the highlighted field (originAdress) in the below response.

[{"streamId":"test","status":"created","playListStatus":null,"type":"liveStream","publishType":null,"name":"test","description":null,"publish":true,"date":1715284969662,"plannedStartDate":0,"plannedEndDate":0,"duration":0,"endPointList":null,"playListItemList":null,"publicStream":true,"is360":false,"listenerHookURL":null,"category":null,"ipAddr":null,"username":null,"password":null,"quality":null,"speed":0.0,"streamUrl":null,**"originAdress":"57.129.35.200"**,"mp4Enabled":0,"webMEnabled":0,"seekTimeInMs":0,"encoderSettingsString":null,"expireDurationMS":0,"rtmpURL":"rtmp://57.129.35.234/LiveApp/test","zombi":false,"pendingPacketSize":0,"hlsViewerCount":0,"dashViewerCount":0,"webRTCViewerCount":0,"rtmpViewerCount":0,"startTime":0,"receivedBytes":0,"bitrate":0,"userAgent":"N/A","latitude":null,"longitude":null,"altitude":null,"mainTrackStreamId":null,"subTrackStreamIds":[],"absoluteStartTimeMs":0,"webRTCViewerLimit":-1,"hlsViewerLimit":-1,"dashViewerLimit":-1,"subFolder":null,"currentPlayIndex":0,"metaData":null,"playlistLoopEnabled":true,"updateTime":0,"autoStartStopEnabled":false,"encoderSettings":null,"anyoneWatching":false}]


As per the recommended way, all streams should be published on the Origin servers and the Edge server should be used only to watch the streams by viewers.

I hope it helps.

**Reply by rsarkisov82** [reply]
Thank you very much for your reply. I see Origin nodes in the response, and it really helps to get the list of published sources. However, is there a way to figure out the Edge nodes from which they receive requests or where these streams are being played?

**Reply by yashtandon113** [reply, marked as answer]
@rsarkisov82 

If you use WebRTC for the playback, then you can pass the random subscriber ID while playing the stream, and then you can get the registered Edge Node IP via the [subscriber list API](https://antmedia.io/rest/#/default/listSubscriberV2). If you use AMS player then you can pass subscriberId like:

`https://domain.antmedia.cloud/LiveApp/play.html?name=test&subscriberId=test
`

Here is the sample API response:

You can find the edge node IP in the highlighted field.

[{"subscriberId":"test","streamId":"test","type":null,"connected":true,"currentConcurrentConnections":134,"concurrentConnectionsLimit":1,"blockedType":null,"blockedUntilUnitTimeStampMs":0,**"registeredNodeIp":"57.129.35.234"**}]

I hope it helps.

**Comment by rsarkisov82** [comment]
I tried to use Subscribers' API requests but they give me a blank response for some reason.  Where do I miss something?

The subscribers use the AMS player, like https://IP:5443/WebRTCAppEE/play.html?name=PROD-KILO03NEW

![ant2](https://github.com/ant-media/Ant-Media-Server/assets/168909671/14878bf1-78ca-433f-8ba9-96810ecf0bc0)
![ant1](https://github.com/ant-media/Ant-Media-Server/assets/168909671/543df672-6ffc-4c20-85f9-477edac7afa9)

**Reply by yashtandon113** [reply]
@rsarkisov82 
I think you missed passing the subscriber ID while playing the stream. It should be something like that.

`https://Domain:5443/WebRTCAppEE/play.html?name=PROD-KILO03NEW&subscriberId=test`

**Comment by rsarkisov82** [comment]
Hi, Let me check if I understood your response correctly.
If a add "...&subscriberId=TEST"  to the AMS player link, I should get a reply reflecting in the API statistics from ...rest/v2/broadcasts/TEST/subscribers/list/0/10  . Correct?
I was playing around with that but unfortunately still couldn't get anything back.
Can you check the screenshots below pls?
![ant4](https://github.com/ant-media/Ant-Media-Server/assets/168909671/372fc303-4cfb-4fa9-a8b2-180248766f76)
![ant3](https://github.com/ant-media/Ant-Media-Server/assets/168909671/3b8ff785-0f5d-422f-978b-cc6d008fe217)

**Reply by yashtandon113** [reply]
@rsarkisov82 

In the API call, the highlighted part should be streamId and not subscriberId.

rest/v2/broadcasts/**streamId**/subscribers/list/0/10

**Reply by rsarkisov82** [reply]
it tested it with stream id also and it made no change. is there anything else should  be configured to make it work?

![ant6](https://github.com/ant-media/Ant-Media-Server/assets/168909671/9a0b3bb7-6b04-4107-823a-e0f5057575aa)
![ant5](https://github.com/ant-media/Ant-Media-Server/assets/168909671/a2485406-11e4-48de-8be8-08a9d7dba1d6)

**Reply by yashtandon113** [reply]
@rsarkisov82 
Which version of AMS are you using? Also, is it possible if you can share your AMS web panel credentials via email at yash.tandon@antmedia.io so I can test further?

**Reply by yashtandon113** [reply]
@rsarkisov82 
Thanks for sharing the details over email.

As you are using v2.5.3 of AMS, which is quite old, this subscriber ID solution will not work for you. So we recommend you upgrade your cluster to version 2.9.0 of AMS.

**Reply by rsarkisov82** [reply]
I sent you another email about it. pls take a look when you have a chance. thank you
- Roman
